Sources/KituraOpenAPI/KituraOpenAPI.swift

```diff
@@ -50,20 +58,56 @@ public class KituraOpenAPI {
             response.status(.OK)
             try response.send(json).end()
         }
+            
         Log.info("Registered OpenAPI definition on \(path)")
     }
 
     private static func addSwaggerUI(to router: Router, with config: KituraOpenAPIConfig) {
-        guard let path = config.swaggerUIPath else {
+        guard let uiPath = config.swaggerUIPath else {
             Log.verbose("No path for SwaggerUI")
             return
         }
+
+        guard let apiPath = config.apiPath else {
+            Log.verbose("No path for OpenAPI definition")
+            return
+        }
+
         guard let sourcesDirectory = Utils.localSourceDirectory else {
             Log.error("Could not locate local source directory")
             return
         }
 
-        router.all(path, middleware: StaticFileServer(path: sourcesDirectory + "/swaggerui"))
-        Log.info("Registered SwaggerUI on \(path)")
+        var aPath = apiPath 
+        if aPath.hasPrefix("/") == false {
+            aPath = "/" + aPath
+        }
+
+        var uPath = uiPath 
+        if uPath.hasPrefix("/") == false {
+            uPath = "/" + uPath
+        }
+
+        let installationRoute = uPath.hasSuffix("/") ? uPath + "assets" : uPath + "/assets"
+        let template = "index.stencil"
+        let swaggerUIInstallation = sourcesDirectory + "/swaggerui"
+
+        let fsLoader = FileSystemLoader(paths: [Path(swaggerUIInstallation)])
+        let environment = Environment(loader: fsLoader)
+
+        let context = ["openapi": aPath, "installation": installationRoute]
+
+        if let rendered = try? environment.renderTemplate(name: template, context: context) {
+            router.get(uPath) { request, response, next in
+                response.send(String(describing: rendered))
```

Comment:
Can we use the new Codable templating we added in Kitura 2.4? https://developer.ibm.com/swift/2018/05/31/type-safe-templating/

Reply:
I had a play with this.  There's a couple of problems:

1. The template we are trying to serve is within the `Kitura-OpenAPI` dependency, not within the user's application, so the template is not in the usual place (`./Views`). And we don't want to mess with the user's `Router.viewsPath` since that would break their own templating.
2. By using the KituraStencil templating (where we register a template engine), we're required to register a StencilTemplateEngine with the user's Router - which could conflict with their own code.